_drafts/ble-throughput-primer.md

```diff
@@ -0,0 +1,288 @@
+
+{: #ble-radio}
+
+### Bluetooth Baseband / Radio
+
+For Bluetooth 4.0, the BLE Radio is capable of transmitting 1 symbol per microsecond and one bit of data can be encoded in each symbol. This gives a raw radio bitrate of 1 Megabit per second (**Mbps**). This is **not** the throughput which will be observed for several reasons:
+
+1. There is a mandatory _150μs_ delay that must be between each packet sent. This is known as the _Inter Frame Space_ (_T_IFS_)
+2. BLE is a _reliable transport_ meaning every packet of data sent from one side must be acknowledged (ACK'd) by the other. The size of an _ACK_ packet is _80bits_ and thus takes _80μs_ to transmit.
+3. The BLE protocol has overhead for every data payload which is sent so sometime is spent sending headers, etc for data payloads
+
+> Fun Fact: As an antenna is used it heats up. As it heats up, the frequency it transmits or receives at can start to drift. The IFS allows for simpler antenna design for BLE by giving the antenna a chance to cooldown before it needs to transmit again
+
+{: #ll-packet}
+
+### The Link Layer (LL) Packet
+
+All data during a BLE connection is sent via _Link Layer_ (**LL**) packets. All higher level messages are packed within _Data Payloads_ of _LL Packets_. Below is what a LL Packet sending data looks like (each tick mark represents 1 byte):
+
+![Data exchange](img/ble-throughput/ll_packet.png)
+
+> NOTE: The astute reader may note that Data Payload can be up to 251 bytes. This however is an optional feature known as "LE Data Packet Length Extension" which we will explore in more detail [below](#ll-data-packet-length-extension)
+
+{:.no_toc}
+
+#### Maximum Link Layer Data Payload Throughput
+
+With the three rules we mentioned about transmissions on the Bluetooth Radio in mind, let's take a look at the procedure to transmit a maximally sized LL packet.
+
+- Side A sends a maximum size LL data packet (**27** bytes of data in a **41** byte payload) which takes **328μs** (`41 bytes * 8 bits / byte * 1Mbps`) to transmit
+- Side B receieves the packet and waits `T_IFS` (**150μs**)
+- Side B ACKs the LL packet it received in **80μs** (**0** bytes of data)
+- Side A waits `T_IFS` before sending any more data
+
+Here's an example exchange of two packets of data in one _Connection Event_:
+![Data exchange](img/ble-throughput/connection_event.png)
+
+The time it takes to transmit one packet can be computed as:
+
+`328μs data packet + 150μs T_IFS + 80μs ACK + 150μs T_IFS = 708μs`
+
+During this time period, **27** bytes of actual data can be transmitted which takes **216μs**.
+
+This yields a raw data throughput of:
+
+`(216μs / 708μs) * 1Mbps = 305,084 bits/second = ~0.381 Mbps`
+
+### L2CAP Channels & Payloads
+
+L2CAP is the higher level protocol packed inside the **Data Payloads** of the Link Layer packets. L2CAP allows:
+
+- unrelated data flows to be multiplexed across different virtual _channels_
```

Comment:
nit: spec talks about "logical channels"

maybe add "(not radio/physical channels)"